streaming/index.js

```diff
@@ -1177,7 +1177,7 @@ const startServer = async () => {
     const location = url.parse(req.url, true);
 
     req.requestId = uuid.v4();
-    req.remoteAddress =  req.headers['x-forwarded-for'].split(',')[0].trim();
+    req.remoteAddress =  req.headers['x-forwarded-for'].split(',')[0].trim() || req.connection.remoteAddress;
```

Comment:
You should be able to delegate this to Express: https://stackoverflow.com/a/14631683

Reply:
Updated.

Reply:
The WebSocket module isn't receiving an express request as `req`, it's a `http.IncomingMessage`, as the WebSocket Server instance is binding to the http server, not the express application:
- https://github.com/mastodon/mastodon/blob/main/streaming/index.js#L174
- https://github.com/mastodon/mastodon/blob/main/streaming/index.js#L891 

So you need to rely on lower-level node.js semantics here, afaik.

`request.connection` on `http.IncomingMessage` did once exist, but it's been deprecated since Node.js 13, superseded by `request.socket` which is the underlying `net.Socket` instance. (this is also the same as `ws._socket`)